aiosmtpd/testing/helpers.py

```diff
@@ -26,59 +26,55 @@ def reset_connection(client):
     # };
     #
     # Is this correct for Windows/Cygwin and macOS?
-    struct_format = 'hh' if sys.platform == 'win32' else 'ii'
+    struct_format = "hh" if sys.platform == "win32" else "ii"
     l_onoff = 1
     l_linger = 0
     client.sock.setsockopt(
         socket.SOL_SOCKET,
         socket.SO_LINGER,
-        struct.pack(struct_format, l_onoff, l_linger))
+        struct.pack(struct_format, l_onoff, l_linger),
+    )
     client.close()
 
 
-# For integration with flufl.testing.
-
-def setup(testobj):
-    testobj.globs['resources'] = ExitStack()
-
-
-def teardown(testobj):
-    testobj.globs['resources'].close()
-
-
-def make_debug_loop():
-    loop = asyncio.get_event_loop()
-    loop.set_debug(True)
-    return loop
+SUPPORTED_COMMANDS_TLS: bytes = (
+    b"Supported commands: AUTH DATA EHLO HELO HELP MAIL "
+    b"NOOP QUIT RCPT RSET STARTTLS VRFY"
+)
 
+SUPPORTED_COMMANDS_NOTLS = SUPPORTED_COMMANDS_TLS.replace(b" STARTTLS", b"")
 
-def start(plugin):
-    if plugin.stderr:
-        # Turn on lots of debugging.
-        patch('aiosmtpd.smtp.make_loop', make_debug_loop).start()
-        logging.getLogger('asyncio').setLevel(logging.DEBUG)
-        logging.getLogger('mail.log').setLevel(logging.DEBUG)
-        warnings.filterwarnings('always', category=ResourceWarning)
 
+def get_server_context() -> ssl.SSLContext:
+    context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+    context.check_hostname = False
+    context.load_cert_chain(
+        resource_filename("aiosmtpd.tests.certs", "server.crt"),
+        resource_filename("aiosmtpd.tests.certs", "server.key"),
+    )
+    return context
 
-def assert_auth_success(testcase: TestCase, code, response):
-    testcase.assertEqual(code, 235)
-    testcase.assertEqual(response, b"2.7.0 Authentication successful")
 
+def get_client_context() -> ssl.SSLContext:
+    context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+    context.check_hostname = False
+    context.load_verify_locations(
+        resource_filename("aiosmtpd.tests.certs", "server.crt")
+    )
+    return context
 
-def assert_auth_invalid(testcase: TestCase, code, response):
-    testcase.assertEqual(code, 535)
-    testcase.assertEqual(response, b'5.7.8 Authentication credentials invalid')
 
+class ReceivingHandler:
```

Comment:
This class is used in many test modules, so it gets consolidated/centralized here.